web/src/lib/domain-logic/getXrayRenewalEvent.ts

```diff
@@ -0,0 +1,25 @@
+import { XrayRegistrationCalendarEvent } from "@businessnjgovnavigator/shared/calendarEvent";
+import { XrayData } from "@businessnjgovnavigator/shared/xray";
+import dayjs from "dayjs";
+
+export const getXrayRenewalEvent = (
+  xrayRegistrationData: XrayData | undefined,
+  year: number,
+  month?: number,
+): XrayRegistrationCalendarEvent | undefined => {
```

Comment:
Pebble - for getLicenseCalendarEvents we always return an array, and don't need to deal with a returned value of undefined. Less headache in my opinion. I'd do something similar here. An array may not be the correct thing. But it would be easier to handle cuz you can use the spread operator

```  
const events: LicenseCalendarEvent[] = [];

  if (licenseData === undefined) {
    return events;
  }
```